services/background-worker/src/Wire/BackendDeadUserNotificationWatcher.hs

```diff
@@ -0,0 +1,98 @@
+{-# LANGUAGE BlockArguments #-}
+
+module Wire.BackendDeadUserNotificationWatcher where
+
+import Cassandra
+import Control.Error.Util (hush)
+import Control.Monad.Trans.Maybe
+import Data.ByteString.Conversion
+import Data.Domain
+import Data.Id
+import Data.Text qualified as Text
+import Data.Text.Encoding qualified as Text
+import Imports
+import Network.AMQP qualified as Q
+import Network.AMQP.Extended
+import Network.AMQP.Lifted qualified as QL
+import UnliftIO
+import Wire.BackgroundWorker.Env
+
+type DomainConsumerMap = IORef (Map Domain (Q.ConsumerTag, MVar ()))
+
+startWatcher :: DomainConsumerMap -> Q.Channel -> AppT IO ()
+startWatcher _consumersRef chan = do
+  markAsWorking BackendDeadUserNoticationWatcher
+
+  cassandra <- asks (.cassandra)
+
+  -- This ensures that we receive notifications 1 by 1 which ensures they are
+  -- delivered in order.
+  lift $ Q.qos chan 0 1 False
+
+  -- TODO: replace bare string with the constant from the rabbitmq PR.
+  let queueName = "dead-user-notifications"
+
+  void . lift $ Q.declareQueue chan Q.newQueue {Q.queueName = queueName}
+  void $ QL.consumeMsgs chan queueName Q.Ack $ \(_msg, envelope) ->
+    let vals = Text.splitOn "." envelope.envRoutingKey
+     in case vals of
+          [uidText, cidText] -> do
+            m <- runMaybeT $ do
+              uid <- hoistMaybe $ hush $ parseIdFromText uidText
+              cid <- hoistMaybe $ fromByteString $ Text.encodeUtf8 cidText
+              pure (uid, cid)
+            (uid, cid) <- maybe handleParseErrors pure m
+            markAsNeedsFullSync cassandra uid cid
```

Comment:
We're likely to get transient notifications here, we should ignore them.

Reply:
What do you mean?

Reply:
We push transient messages with ttl of 0. This probably means that most of them get dead lettered, unless there is an active consumer at that point. But we shouldn't count these as missed notifications because they're meant to be missed if the client is not online. 

Another idea would be to make them go to some other DLX which just drops them, so we don't have to worry about it here.

They're pushed here: https://github.com/wireapp/wire-server/blob/b74b7e67abd4f3f603c3b7b956e535577e4e853e/services/gundeck/src/Gundeck/Push.hs#L296-L307